packages/aws-cdk-lib/aws-eks/lib/cluster.ts

```diff
@@ -1869,6 +1869,20 @@ export class Cluster extends ClusterBase {
     });
   }
 
+  /**
+   * Manually associate an `OpenIdConnectProvider` resource to this cluster. Skips the lazy
+   * allocation of an OpenIdConnectProvider.
+   *
+   * @param arn the arn of the OIDC provider
+   */
+  public associateOpenIdConnectProvider(arn: string): iam.IOpenIdConnectProvider {
```

Comment:
This would need to accept a `IOpenIdConnectProvider`. not an ARN.

Reply:
For sure, I was pretty new to the CDK and my exact use case when I first made this PR. Still finishing up some more important work but once I finish I'm going to clean this up and finish off the parts the PR is missing.